### PR TITLE
Adding some examples of generated code

### LINKED ISSUE #9: Add code examples for the generated code
The project received a lot of feedbacks related to "how the generated code will look like". @jbrick Suggested we should add a section with _Example of Generated code_. Also this should highlight that we generate both Retrofit Apis and Kotlin Data classes (as is not really visible from the README).

### PR DESCRIPTION
**Pull request by cortinico**
Fixes #9 

Since several users requested to see the generated code, I'm adding some examples to the repo (a project without the `.gitignore`) plus a section to the README file.

### CODE REVIEW

**Comment by cortinico** [comment]
@macisamuele the table in the readme is not as great as I expected :| https://github.com/cortinico/swagger-gradle-codegen/tree/9-fix-code-examples

Should we probably "stack" the examples?

File: README.md [hunk 1]
```diff
@@ -63,6 +63,136 @@ Here the list of the supported platforms:
 
 We're looking forward to more platforms to support in the future. Contributions are more than welcome.
 
+## Examples
+
+You can find some **examples** in this repository to help you set up your generator environment.
+
+* [sample-generated-code](/sample-generated-code) Contains an example of an Android Library where the code is generated inside the `/scr/main/java` folder. You can use this example to see how the generate code will _look like_ (like [here](/sample-generated-code/src/min/java/com/yelp/codegen/generatecodesamples/apis/ResourceApi.kt))
```
Comment (olivierperez) [inline README.md]:
```suggestion
* [sample-generated-code](/sample-generated-code) Contains an example of an Android Library where the code is generated inside the `/scr/main/java` folder. You can use this example to see how the generated code will _look like_ (like [here](/sample-generated-code/src/min/java/com/yelp/codegen/generatecodesamples/apis/ResourceApi.kt))
```

File: README.md [hunk 2]
```diff
@@ -63,6 +63,136 @@ Here the list of the supported platforms:
...
+
+* [sample-groovy-android](/sample-groovy-android) Contains an example of an Android Library configured with a `build.gradle` file, using Groovy as scripting language.
+
+* [sample-kotlin-android](/sample-kotlin-android) Contains an example of an Android Library configured with a `build.gradle.kts` file, using Kotlin as scripting language.
+
+## How the generated code will look like
+
+Here you can find some example of how the generated code will look like in your project:
```
Comment (olivierperez) [inline README.md]:
```suggestion
Here you can find some examples of how the generated code will look like in your project:
```

File: README.md (hunk 1 above)
Comment (cortinico) [reply README.md]: Thanks, fixed! 👍

File: README.md (hunk 2 above)
Comment (cortinico) [reply README.md]: Thanks, fixed! 👍

File: README.md [hunk 3]
```diff
@@ -63,6 +63,20 @@ Here the list of the supported platforms:
 
 We're looking forward to more platforms to support in the future. Contributions are more than welcome.
 
+## Examples
+
+You can find some **examples** in this repository to help you set up your generator environment.
+
+* [sample-generated-code](/sample-generated-code) Contains an example of an Android Library where the code is generated inside the `/scr/main/java` folder. You can use this example to see how the generated code will _look like_ (like [here](/sample-generated-code/src/min/java/com/yelp/codegen/generatecodesamples/apis/ResourceApi.kt))
```
Comment (macisamuele) [inline README.md]: the link appears broken

File: README.md [hunk 4]
```diff
@@ -63,6 +63,20 @@ Here the list of the supported platforms:
...
+
+* [sample-groovy-android](/sample-groovy-android) Contains an example of an Android Library configured with a `build.gradle` file, using Groovy as scripting language.
```
Comment (macisamuele) [inline README.md]: nit. I would not mention the following ", using Groovy as scripting language"

File: README.md [hunk 5]
```diff
@@ -63,6 +63,20 @@ Here the list of the supported platforms:
...
+
+* [sample-kotlin-android](/sample-kotlin-android) Contains an example of an Android Library configured with a `build.gradle.kts` file, using Kotlin as scripting language.
```
Comment (macisamuele) [inline README.md]: as above

File: SAMPLES.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+
+## `kotlin` platform
+
+### Data Class
+
+The following definition:
+
+```json
+"definitions": {
+        "Pet": {
```
Comment (macisamuele) [inline SAMPLES.md]: Please keep indentation uniform (now you have 8 spaces between `definitions` and `Pet` and 4 spaces everywhere-else)

File: SAMPLES.md [hunk 7]
```diff
@@ -0,0 +1,147 @@
+In this page you can find some samples of generated code using swagger-gradle-codegen for every platform:
+
+## `kotlin` platform
+
+### Data Class
+
+The following definition:
```
Comment (macisamuele) [inline SAMPLES.md]: I would suggest to reorganise it a bit in order to have a complete example of swagger specs

You could have the specs at the end of the page as reference for the kotlin code that you're representing

File: SAMPLES.md [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+)
+```
+
+### Retrofit Interface
+
+The following path:
+
+```
+"paths": {
+        "/pet/{petId}": {
```
Comment (macisamuele) [inline SAMPLES.md]: as above

File: SAMPLES.md [hunk 9]
```diff
@@ -0,0 +1,147 @@
...
+@JvmSuppressWildcards
+interface DefaultApi {
+    /**
+     * Find pet by ID
+     * Returns a single pet
+     * The endpoint is owned by generatecodesamples service owner
+     * @param petId ID of pet to return (required)
+     */
+    @Headers(
+            "X-Operation-ID: getPetById"
```
Comment (macisamuele) [inline SAMPLES.md]: weird indentation

File: README.md (hunk 4 above)
Comment (cortinico) [reply README.md]: I've rephrased a bit

File: SAMPLES.md (hunk 7 above)
Comment (cortinico) [reply SAMPLES.md]: Done, now the specs are on top :)

**Pull request merged**